### PR TITLE
Fix OTHER_LDFLAGS regression

### LINKED ISSUE #8333: 1.6.0 fails to set podspec libraries in pod target's OTHER_LDFLAGS
* [ X] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

`git clone git@github.com:paulb777/CocoaPodsMissingLibs.git`
`pod install`
Build target in Xcode

## What did you expect to happen?

Successful link

## What happened instead?

[Link failure because of missing libz symbols](https://github.com/grpc/grpc/issues/17365)

It successfully links when installed with CocoaPods 1.5.3.

```
$ diff "./1.5.3-Pods/Target Support Files/gRPC-Core/gRPC-Core.xcconfig" "./Pods/Target Support Files/gRPC-Core/gRPC-Core.xcconfig"
8d7
< OTHER_LDFLAGS = -l"c++" -l"z"
```

### Stack

```
   CocoaPods : 1.6.0.beta.2
        Ruby : ruby 2.4.4p296 (2018-03-28 revision 63013) [x86_64-darwin17]
    RubyGems : 2.7.8
        Host : Mac OS X 10.13.6 (17G2112)
       Xcode : 10.1 (10B61)
         Git : git version 2.19.0.605.g01d371f741-goog
Ruby lib dir : /Users/paulbeusterien/.rvm/rubies/ruby-2.4.4/lib
Repositories : cpdc-internal-spec - sso://cpdc-internal/spec @ 152c9886b33dd7f4fd66ae9b1263bf08290815e0
               master - https://github.com/CocoaPods/Specs.git @ 46ff06ff0d46bc5934fc18283849b487bf095450
               paul-specs - git@github.com:paulb777/Specs.git @ 93533c19dbfdd1f4705d38fe57a3407a8b279d07
               paulb777 - https://github.com/paulb777/Specs.git @ 93533c19dbfdd1f4705d38fe57a3407a8b279d07
```

### Installation Source

```
Executable Path: /Users/paulbeusterien/CocoaPods/CocoaPods/bin/pod
```

### Plugins

```
claide-plugins         : 0.9.2
cocoapods-deintegrate  : 1.0.2
cocoapods-dependencies : 1.0.0.beta.1
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.0.0
cocoapods-trunk        : 1.3.1
cocoapods-try          : 1.1.0
cocoapods_debug        : 0.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'MissingLibs' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  use_frameworks!

  # Pods for MissingLibs
  pod 'gRPC-Core'

end
```

## Project that demonstrates the issue

https://github.com/paulb777/CocoaPodsMissingLibs

### PR DESCRIPTION
**Pull request by paulb777**
Fix #8333 - a regression introduced in #8314 

Let me know if this is reasonable and I'll update the integration tests

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix OTHER_LDFLAGS regression  
  [paulb777](https://github.com/paulb777)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
@paulb777 in 1.5.3 does the aggregate target contain the system libraries/frameworks?

**Comment by dnkoutso** [comment]
so thats weird no? How was 1.5.3 working? Where was it getting these flags from?

**Comment by paulb777** [comment]
Presumably, in 1.5.3, the library references were built into the library during the pod target build.

**Comment by paulb777** [comment]
The aggregate target links as a static library so it doesn't need to resolve the references.

File: spec/unit/target/build_settings/pod_target_settings_spec.rb [hunk 1]
```diff
@@ -78,7 +78,7 @@ class BuildSettings
           end
 
           it 'does not include the libraries for the specifications' do
```
Comment (sebastianv1) [inline spec/unit/target/build_settings/pod_target_settings_spec.rb]: nit: Change to `does include`

**Comment by dnkoutso** [comment]
Would like one more approval here. After this we can ship 1.6.0.RC

File: lib/cocoapods/target/build_settings.rb [hunk 2]
```diff
@@ -629,9 +629,9 @@ def initialize(target, non_library_spec = nil)
           libraries = []
           if non_library_xcconfig? || target.build_as_dynamic?
             libraries.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
+            libraries.concat libraries_to_import
```
Comment (segiddins) [inline lib/cocoapods/target/build_settings.rb]: should this only be dynamic libraries ?

Comment (paulb777) [reply lib/cocoapods/target/build_settings.rb]: I believe its for all libraries. It's only for building a dynamic library, but it needs to know all of its static and dynamic library dependencies.

Comment (segiddins) [reply lib/cocoapods/target/build_settings.rb]: Does this mirror what we have for frameworks?

Comment (paulb777) [reply lib/cocoapods/target/build_settings.rb]: Yes, frameworks always get added to `OTHER_LDFLAGS`.

**Comment by segiddins** [comment]
👍🏻

**Pull request merged**